src/main/java/org/rulez/demokracia/pdengine/BeatTable.java

```diff
@@ -6,18 +6,18 @@
 import org.rulez.demokracia.pdengine.dataobjects.Pair;
 import org.rulez.demokracia.types.MapMatrix;
 
-public class BeatTable extends MapMatrix<Choice, Pair> implements ContainingBeats{
+public class BeatTable extends MapMatrix<String, Pair> implements ContainingBeats{
```

Comment:
What is the reason, why you changed this type?

Reply:
There was a design error in the specification. We agreed with @magwas in this solution. :)

Reply:
Open for extension close for modification...
I'm not sure I exactly understand what was the design mistake, but what if you create a superclass for Choice that contains a single String and use it to don't cause tons of conflicts?

Reply:
@magwas opinion?

Reply:
My observation was this: 
As far as I understand the ticket. I have to implement a BeatTable normalizeBeatMatrix(List castVotes); function. I checked the CastVote class and the RankedChoice class too. However I just found choiceId in the RankedChoice. So I can create an order based on the RankedChoices rank but only for the choiceIds. The problem is that the BeatTables matrix contains choices.
So my question is that how can I create a full valid Choice from a choiceId? Where can I find the relation between choiceIds and Choice objects?
@magwas : 
Oops, a design inconsistency again...
Can you refactor BeatTable to use choice ID instead of Choice object?

Reply:
@szirbucz no need to keep backward compatibility, that would only increase code complexity with no gain. It was an error to store references to choices in different ways in different places.
the open/closed principle is more about the design of behaviour of a class, isn't it?

Reply:
So if you are sure, that we won't need anything else but string for key, it might be ok, but in any other cases I feel it a bad choice to use a final class. IMHO Serializable would be more than enough constraint for the key type.
Open-Close is about the behavior, but the used types can also affect the behavior. If we want to use some other type than String for key type, we will have a bad time.

Reply:
I see your point. My point is that the type should be the same across the board.
It is arguable that maybe changing strings in other places to choice would have been better.
But this class is very much tied to this particular implementation, so no need to generalize.
The TDD answer to the question of "how much we should generalize?" is that we should defer design choices: when the need to generalize comes, we generalize. refactor is cheap, we are doing TDD.